File: pets/qsort.php
<?php

function qsort(array $arr): array {
  if(count($arr) < 2){
    return $arr;
  } else {
    $half_index = count($arr) / 2;
    $pivot = $arr[(int)$half_index];

    $less_than = [];
    $more_than = [];

    foreach ($arr as $key => $value) {
      if($value < $pivot){
        $less_than[] = $value;
      } elseif ($value > $pivot) {
        $more_than[] = $value;
      }
    }

    $less_than = qsort($less_than);
    $more_than = qsort($more_than);

    return array_merge($less_than,array($pivot),$more_than);
  }
}

$arr = [64,13,22,16,4,5];

var_dump(qsort($arr));

File: pets/qsort.go
package main

import (
	"fmt"
)

func qsort(arr []int) []int {
	if len(arr) < 2 {
		return arr
	} else {
		//Вычисляем серидину массива
		half_index := len(arr) / 2

		//Записываем наш опорный индекс
		pivot := arr[half_index]

		less_than := []int{}
		more_than := []int{}

		for _, value := range arr {
			if value < pivot {
				less_than = append(less_than, value)
			} else if value > pivot {
				more_than = append(more_than, value)
			}
		}

		less_than = qsort(less_than)
		more_than = qsort(more_than)

		result_arr := append(less_than, pivot)
		result_arr = append(result_arr, more_than...)

		return result_arr
	}

	return arr
}

func main() {
	arr := []int{64, 13}

	fmt.Println(qsort(arr))
}
